cmd/lakefs/cmd/migrate.go

```diff
@@ -70,11 +72,53 @@ var upCmd = &cobra.Command{
 		}
 		defer kvStore.Close()
 
-		_ = mustValidateSchemaVersion(ctx, kvStore)
-		fmt.Printf("No migrations to apply.\n")
+		wasMigrated, err := doMigration(ctx, kvStore, cfg)
+		if err != nil {
+			_, _ = fmt.Fprintf(os.Stderr, "Migration failed: %s\n", err)
+			os.Exit(1)
+		}
+		if wasMigrated {
+			fmt.Printf("Migration completed successfully.\n")
+		} else {
+			fmt.Printf("No migrations to apply.\n")
+		}
 	},
 }
 
+func doMigration(ctx context.Context, kvStore kv.Store, cfg *config.Config) (bool, error) {
+	var (
+		version     int
+		wasMigrated bool
+		err         error
+	)
+	for version < kv.LatestVersion {
+		version, err = kv.ValidateSchemaVersion(ctx, kvStore)
+		if err != nil && !errors.Is(err, kv.ErrMigrationRequired) {
+			return false, err
+		}
+		switch {
+		case version < kv.ACLNoReposMigrateVersion:
+			return false, fmt.Errorf("migration to ACL required. Did you migrate using version v0.99.x? https://docs.lakefs.io/reference/access-control-list.html#migrating-from-the-previous-version-of-acls: %w", kv.ErrMigrationRequired)
+
+		case version < kv.ACLImportMigrateVersion:
+			// skip migrate to ACL for users with External authorizations
+			if !cfg.IsAuthUISimplified() {
+				fmt.Println("skipping ACL migration - external Authorization")
+				err = kv.SetDBSchemaVersion(ctx, kvStore, kv.ACLImportMigrateVersion)
+				if err != nil {
+					return false, fmt.Errorf("failed to upgrade version, to fix this re-run migration: %w", err)
+				}
+			} else {
+				if err = migrations.MigrateImportPermissions(ctx, kvStore); err != nil {
+					return false, err
+				}
+			}
+			wasMigrated = true
+		}
+	}
+	return wasMigrated, nil
```

Comment:
We use ValidateSchemaVersion to get the current version, The loop stops only after one iteration because as of now we only have one migration that can be a rolling one. In the future we could potentially have more. The first case is the minimal version required to perform a rolling migration since this requires a specific version